packages/gatsby/src/schema/infer/inference-metadata.js

```diff
@@ -118,55 +153,124 @@ const getType = (value, key) => {
   }
 }
 
-const updateValueDescriptor = (
-  { nodeId, key, value, operation = `add` /* add | del */, descriptor = {} },
-  path = []
+const updateValueDescriptorObject = (
+  value,
+  typeInfo,
+  nodeId,
+  operation,
+  metadata,
+  path
 ) => {
-  // The object may be traversed multiple times from root.
-  // Each time it does it should not revisit the same node twice
-  if (path.includes(value)) {
-    return [descriptor, false]
-  }
+  path.push(value)
 
-  const typeName = getType(value, key)
+  const { dprops = {} } = typeInfo
```

Comment:
Mind adding a comment on what `dprops` name stands for? A descriptor for properties or did you imply something else?

Reply:
Hm I had such a comment. Seems that got lost somewhere. Yeah I'll add it.